crypto/ec/curve448/constant_time.h

```diff
@@ -0,0 +1,337 @@
+    unsigned char maskc;
+# endif
+
+    memset(out, 0, elem_bytes);
+    for (j = 0; j < n_table; j++, big_i -= big_one) {
+        big_register_t br_mask = br_is_zero(big_i);
+        word_t mask;
+
+# ifndef HAS_UNALIGNED_STRUCTS
+        maskc = (unsigned char)br_mask;
+# endif
+
+        for (k = 0; k <= elem_bytes - sizeof(big_register_t);
+             k += sizeof(big_register_t)) {
+            if (elem_bytes % sizeof(big_register_t)) {
+                /* unaligned */
+# ifdef HAS_UNALIGNED_STRUCTS
+                ((unaligned_br_t *)(out + k))->unaligned |=
+                        br_mask
+                        & ((const unaligned_br_t *)
+                           (&table[k + j * elem_bytes]))->unaligned;
+# else
+                size_t i;
+
+                for (i = 0; i < sizeof(big_register_t); i++)
+                    out[k + i] |= maskc
+                                  & ((unsigned char *) table)
+                                    [k + (j * elem_bytes) + i];
+# endif
+            } else {
+                /* aligned */
+                *(big_register_t *)(out + k) |=
+                        br_mask
+                        & *(const big_register_t *)(&table[k + j * elem_bytes]);
+            }
+        }
+
+        mask = word_is_zero(idx ^ j);
+# ifndef HAS_UNALIGNED_STRUCTS
+        maskc = (unsigned char)mask;
+# endif
+        if (elem_bytes % sizeof(big_register_t) >= sizeof(word_t)) {
+            for (; k <= elem_bytes - sizeof(word_t); k += sizeof(word_t)) {
+                if (elem_bytes % sizeof(word_t)) {
+                    /* input unaligned, output aligned */
+# ifdef HAS_UNALIGNED_STRUCTS
+                    *(word_t *)(out + k) |=
+                            mask
+                            & ((const unaligned_word_t *)
+                               (&table[k + j * elem_bytes]))->unaligned;
+# else
+                    size_t i;
+
+                    for (i = 0; i < sizeof(word_t); i++)
+                        out[k + i] |= maskc
+                                      & ((unsigned char *)table)
+                                         [k + (j * elem_bytes) + i];
+# endif
+                } else {
+                    /* aligned */
+                    *(word_t *)(out + k) |=
+                            mask
+                            & *(const word_t *)(&table[k + j * elem_bytes]);
+                }
+            }
+        }
+
+        if (elem_bytes % sizeof(word_t)) {
+            for (; k < elem_bytes; k += 1) {
+                out[k] |= mask & table[k + j * elem_bytes];
+            }
+        }
+    }
+}
+
+/*
+ * Constant-time a = mask ? bTrue : bFalse.
+ *
+ * The input and output must be at least as aligned as alignment_bytes
+ * or their size, whichever is smaller.
+ *
+ * Note that the output is not __restrict__, but if it overlaps either
+ * input, it must be equal and not partially overlap.
+ */
+static ossl_inline void constant_time_select(void *a_,
```

Comment:
at least this function might collide with one of our constant time functions

Reply:
I renamed it